datastacks/config_class.py

```diff
@@ -0,0 +1,48 @@
+from datetime import date
+from typing import List, Optional
+
+from pydantic import BaseModel, Field
+
+
+class IngestConfig(BaseModel):
+    dataset_name: str = Field(
+        description="Dataset name, used to derive pipeline and linked service names."
+    )
+    pipeline_description: str = Field(
+        description="Description of the pipeline to be created."
+    )
+    data_source_type: str = Field(description="Datasource type eg. azure_sql.")
+
+    key_vault_linked_service_name: str = Field(
+        description="Name of the keyvault service to connect to."
+    )
+    data_source_password_key_vault_secret_name: str = Field(
+        description="Secret name of the data source password."
+    )
+    data_source_connection_string_variable_name: str = Field(
+        description="Variable name for the connection string."
+    )
+
+    ado_variable_groups_nonprod: List[str] = Field(
```

Comment:
Since Python 3.9 you don't have to import List from typing, you can just write `list[str]`. Feel free to resolve.

Reply:
Done